conanfile.py

```diff
@@ -91,6 +92,14 @@ def set_version(self):
         version = re.search(r'(\d+\.\d+\.\d+)', content).group(1)
         self.version = version.strip()
 
+    def checkMoldIsSupported(self):
```

Comment:
I've moved the logic to a function here as it's complicated enough we want this to be an atomic unit of functionality.  As you will see in the CMake we will no longer duplicate this logic there too.